### ISSUE TITLE
Not all metrics available in Prometheus

### ISSUE THREAD

**Issue by lkshck** [eko/pihole-exporter#61]
I recognized that just some metrics are available in Prometheus instead of the whole list which is provided here. This causes that some panels from the Dashboard are not working. Is it possible to add those metrics afterwards?

<img width="703" alt="grafik" src="https://user-images.githubusercontent.com/46962619/92434669-b875a300-f1a0-11ea-947f-1eaf3d4d0ceb.png">

**Comment by lkshck** [comment]
I started the exporter now manually and all metrics are available, but if i'm running the exporter as an service, not all metrics are available. 
Service under /lib/systemd/system/pihole_exporter.service looks like the following
`[Unit]
Description=pihole_exporter

[Service]
ExecStart=/opt/pihole_exporter/pihole_exporter-linux-arm
WorkingDirectory=/opt/pihole_exporter
Restart=always
User=pihole_exporter

[Install]
WantedBy=multi-user.target`

Any ideas how this could happen?

**Comment by jpcapone** [comment]
I am experiencing the same exact issue.  Please advise.

**Comment by Bencyril** [comment]
Hi guys !
I had the same issues earlier and tried to understand why, in my case, it's due to the fact that I used Pihole with https and the repo uses scratch as the base for the container. It causes the app to crash due to missing certificates

`An error has occured during retrieving PI-Hole statistics Get "https://xxxxxxxx:443/admin/api.php?summaryRaw&overTimeData&topItems&recentItems&getQueryTypes&getForwardDestinations&getQuerySources&jsonForceObject&auth=xxxxxxx": x509: certificate signed by unknown authority <-------
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x48 pc=0x7a9438]

goroutine 25 [running]:
github.com/eko/pihole-exporter/internal/pihole.(*Client).getStatistics(0xc000020510, 0xc00003c790)
	/go/src/github.com/eko/pihole-exporter/internal/pihole/client.go:163 +0x308
github.com/eko/pihole-exporter/internal/pihole.(*Client).Scrape(0xc000020510)
	/go/src/github.com/eko/pihole-exporter/internal/pihole/client.go:61 +0x83
created by main.initPiHoleClient
	/go/src/github.com/eko/pihole-exporter/main.go:37 +0xc0`

To solved the issues, I added to the Dockerfile ca-certificates package and it fixed the error

**Comment by PleaseStopAsking** [comment]
@lkshck Did you find the proper solution to this? I tried to pass the password via argument but it doesn't seem to make a difference.
```
[Unit]
Description=pihole_exporter

[Service]
ExecStart=/opt/pihole_exporter/pihole_exporter-linux-amd64 "pihole_password example"
WorkingDirectory=/opt/pihole_exporter
Restart=always
User=pihole_exporter

[Install]
WantedBy=multi-user.target
```

**Comment by lkshck** [comment]
@PleaseStopAsking No unfortunately not. After some tries I didn't searched again for a proper solution, but would be interesting if anyone knows how to fix it.

**Comment by PleaseStopAsking** [comment]
@lkshck I ended up getting it to work finally with the below service file contents. I had tried this same thing previously with the difference this time being that I started it with ```sudo service pihole_exporter restart``` instead of ```sudo service pihole_exporter start```. I believe that due to it already running, it was not getting restarted correctly and was not picking up the ```pihole_password``` argument and value.

```                                                       
[Unit]
Description=pihole_exporter

[Service]
ExecStart=/opt/pihole_exporter/pihole_exporter-linux-amd64 "-pihole_password=example"
WorkingDirectory=/opt/pihole_exporter
Restart=always
User=pihole_exporter

[Install]
WantedBy=multi-user.target
```

**Comment by lkshck** [comment]
Will try that, thanks for the update!

**Comment by kquinsland** [comment]
So to confirm, some data is _not_ available to the exporter when using the API, but when using the old password, the data *is* available?

**Comment by PleaseStopAsking** [comment]
@kquinsland Yes and No. Some data is NOT available to the exporter when Pi-Hole credentials are not passed to it so that it can auth into Pi-Hole but when using you pass credentials to the exporter, all available metrics are exported.

Pi-Hole by design provides a limited number of metrics on its dashboard without authentication (API and GUI) but much more if you login.

**Comment by kquinsland** [comment]
@PleaseStopAsking 

Is there a difference in the metrics that can be reported when the exporter is using the API key versus the web password? There are several metrics that the bundled grafana dashboard implies should be there, but I don't see them when i hit the `/metrics` endpoint. I have not used the web password to authenticate, just the API token.

**Comment by PleaseStopAsking** [comment]
@kquinsland Not that I am aware of but maybe someone else can confirm.

**Comment by kquinsland** [comment]
@PleaseStopAsking  Yep, i was able to do some testing. Turns out, the password and API token work the same as far as the exporter is concerned. The reason that I didn't have all the metrics showing up is because I had a mis-configuration on my end and the API token was being passed in incorrectly. As soon as i cleared that up, the full suite of metrics showed up :).

a `WARN` in the logs saying authenticated request didn't contain all expected metrics or a `LOG_LEVEL=trace` setting to dump the full request/payload would be really helpful for tracking this down.  :/

TL;DR: if some metrics are not showing up, check your auth :)

**Comment by eko** [comment]
Indeed, some metrics require you to be authenticated to be scraped from PI-Hole API, I think you have an authentication issue (or the argument in your service declaration is not taken into account).

Closing this issue as it is not related to pihole-exporter. I hope you will find a way to fix this.

Thank you
